### DISCUSSION TITLE
Warn about silent data loss with get_to

### DISCUSSION THREAD

**Discussion by provtemp** [nlohmann/json#3413] [Ideas]
### Description

I think the documentation for [get_to](https://json.nlohmann.me/api/basic_json/get_to/) should warn users that they will *not* get a runtime exception when the sink cannot hold the value.

```c++
    auto v = json::parse(R"({"big":4294967296})");
    int32_t big;

    if (v["big"] == 0) {
      throw "big must not be zero";

    } else {
      v["big"].get_to(big);
    }

    std::cout << "big is " << big << '\n';
```

Here `big` ends up being zero even though the code tries to prevent that. This can have security implications for some applications. 

### Reproduction steps

.

### Expected vs. actual results

.

### Compiler and operating system

.

### Library version

.

### Validation

- [ ] The bug also occurs if the latest version from the [`develop`](https://github.com/nlohmann/json/tree/develop) branch is used.
- [ ] I can successfully [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests).

**Comment by gregmarr** [comment]
I could see adding a note about this, but I'm curious why someone would think that they *would* get a runtime exception in this case.  I don't see anything in the documentation that says that there would be one.  Was it just an assumption that there would be, or is there some other documentation that would lead someone to believe that and should also be changed?
